containers/src/Kokkos_UnorderedMap.hpp

```diff
@@ -295,17 +295,7 @@ class UnorderedMap {
   //! \name Public member functions
   //@{
 
-  UnorderedMap()
-      : m_bounded_insert(),
-        m_hasher(),
-        m_equal_to(),
-        m_size(),
-        m_available_indexes(),
-        m_hash_lists(),
-        m_next_index(),
-        m_keys(),
-        m_values(),
-        m_scalars() {}
+  UnorderedMap() : UnorderedMap(0) { m_bounded_insert = false; }
```

Comment:
Please elaborate on why you flipped the `m_bounded_insert` data member and not just delegated to the other constructor.

Reply:
To me, this is the only real difference between the constructors. So I am flipping it back to retain the original behavior.

Reply:
I doubt there was much thought put into that previous version of the default constructor.  My preference would go for giving a default value to the first argument of the other constructor.
Let's see what other think.

Reply:
Since I wrote UnorderedMap I can verify that we didn't give much thought to the values of a default constructed map, since a default constructed map was designed to be an invalid map (similar to how a default constructed View is invalid).  In general I like this change because having an invalid default map doesn't buy us anything.  However, don't set `m_bounded_insert = false;`.  This will slow the code down greatly if you resize the map instead of assigning a new map to it because every thread will search the entire map to find an empty space before failing (greatly increasing the atomic contention).

Reply:
OK, thanks! I will change it accordingly.